server/src/main/scala/io/delta/sharing/server/model.scala

```diff
@@ -135,3 +135,130 @@ object Action {
   // Basically delta sharing doesn't support write for now.
   val maxWriterVersion = 0
 }
+
+/**
+ * Actions defined in delta format, used in response when requested format is delta.
+ */
+
+sealed trait DeltaAction {
+  /** Turn this object to the [[DeltaSingleAction]] wrap object. */
+  def wrap: DeltaSingleAction
+}
+
+/**
+ * Used to block older clients from reading the shared table when backwards
+ * incompatible changes are made to the protocol. Readers and writers are
+ * responsible for checking that they meet the minimum versions before performing
+ * any other operations.
+ *
+ * Since this action allows us to explicitly block older clients in the case of a
+ * breaking change to the protocol, clients should be tolerant of messages and
+ * fields that they do not understand.
+ */
+case class DeltaProtocol(minReaderVersion: Int) extends DeltaAction {
+  override def wrap: DeltaSingleAction = DeltaSingleAction(protocol = this)
+}
+
+/**
+ * DeltaAddFile used in delta sharing protocol, copied from AddFile in delta.
+ *   Adding 4 delta sharing related fields: id/version/timestamp/expirationTimestamp,
+ *       which will be redacted at the client side before writing to the delta log.
+ *   Ignoring 1 field: tags.
+ */
+case class DeltaAddFile(
+    path: String,
+    id: String,
```

Comment:
Question: why do we want the server to generate this id? It adds load on the server. Can the client generate the id as needed?

Reply:
I think it is fine to have it in the server. Just tested this and we generate these at 2M/sec.

Reply:
thanks for checking the rate!